sandbox/src/PersonalizationAnalyticsClientSide.js

```diff
@@ -0,0 +1,245 @@
+    });
+
+    sendEvent({
+      eventType: "decisioning.propositionDisplay",
+      executedPropositions
+    });
+
+    getECID(instanceName).then(visitorID => {
+      const analyticsPayload = concatinateAnalyticsPayloads(analyticsPayloads);
+      triggerAnalyticsHit({ analyticsPayload, visitorID });
+    });
+  });
+};
+
+const sendEvent = ({
+  eventType,
+  viewName,
+  decisionScopes,
+  renderDecisions,
+  executedPropositions
+}) => {
+  const xdm = {
+    eventType: eventType
+  };
+
+  if (viewName) {
+    xdm.web = {
+      webPageDetails: {
+        viewName
+      }
+    };
+  }
+
+  if (executedPropositions) {
+    xdm._experience = {
+      decisioning: {
+        propositions: executedPropositions
+      }
+    };
+  }
+
+  return window[instanceName]("sendEvent", {
+    renderDecisions,
+    decisionScopes,
+    xdm
+  });
+};
+
+const Products = () => {
+  sendEvent({
+    eventType: "view-change",
+    viewName: "products",
+    renderDecisions: true
+  }).then(result => {
+    if (!result.propositions) {
+      return;
+    }
+    const analyticsPayload = collectAnalyticsPayloadData(result.propositions);
+    getECID(instanceName).then(visitorID => {
+      triggerAnalyticsHit({ analyticsPayload, visitorID });
+    });
+  });
+  return (
+    <div>
+      <h2>Products</h2>
+      <div
+        style={{ border: "1px solid red" }}
+        id="personalization-products-container"
+      >
+        This is the personalization placeholder for the products view.
+        Personalized content has not been loaded.
+      </div>
+    </div>
+  );
+};
+
+const Cart = () => {
+  sendEvent({
+    eventType: "view-change",
+    viewName: "cart",
+    renderDecisions: true
+  }).then(result => {
+    if (!result.propositions) {
+      return;
+    }
+    const analyticsPayload = collectAnalyticsPayloadData(result.propositions);
+    getECID(instanceName).then(visitorID => {
+      triggerAnalyticsHit({ analyticsPayload, visitorID });
+    });
+  });
+
+  return (
+    <div>
+      <h2>Cart</h2>
+      <div
+        style={{ border: "1px solid red" }}
+        id="personalization-cart-container"
+      >
+        This is the personalization placeholder for the cart view. Personalized
+        content has not been loaded.
+      </div>
+    </div>
+  );
+};
+
+export default function PersonalizationAnalyticsClientSide() {
+  useEffect(() => {
+    const xdm = {};
+    xdm.eventType = "page-view";
+
+    window[instanceName]("sendEvent", {
+      renderDecisions: true,
+      xdm
+    }).then(result => {
+      if (!result.propositions) {
+        return;
+      }
+
+      const analyticsPayload = collectAnalyticsPayloadData(result.propositions);
+      getECID(instanceName).then(visitorID => {
+        triggerAnalyticsHit({ analyticsPayload, visitorID });
+      });
+    });
+  }, []);
+
+  const match = useRouteMatch();
+
+  return (
+    <div>
+      <ContentSecurityPolicy />
+      <h1>Personalization with A4T client side logging</h1>
```

Comment:
Maybe we should add a bit more information here about this specific use-case (using the fetch call to the analytics end-point etc.). And link to documentation that describes it.